wa/output_processors/postgresql.py

```diff
@@ -0,0 +1,601 @@
+                        (
+                            self.current_classifier_uuid,
+                            None,
+                            self.current_metric_uuid,
+                            classifier,
+                            metric.classifiers[classifier]))
+                self.metrics_already_added.append(metric)
+            # Run Augmentations initial population
+            for augmentation in run_output.augmentations:
+                self.current_augmentation_uuid = uuid.uuid4()
+                self.cursor.execute(
+                    self.sql_command['create_augmentation'],
+                    (
+                        self.current_augmentation_uuid,
+                        self.run_uuid,
+                        augmentation))
+                self.insert_parameters(
+                    'augmentation',
+                    run_output.run_config.augmentations[augmentation],
+                    owner_id=self.current_augmentation_uuid,
+                    job_specific=False)
+                self.augmentations_already_added[augmentation] = self.current_augmentation_uuid
+            # Targets table (one time)
+            self.cursor.execute(
+                self.sql_command['create_target'],
+                (
+                    self.target_uuid,
+                    self.run_uuid,
+                    target_info.target,
+                    target_info.cpus,
+                    target_info.os,
+                    target_info.os_version,
+                    target_info.hostid,
+                    target_info.hostname,
+                    target_info.abi,
+                    target_info.is_rooted,
+                    # Important caveat: kernel_version is the name of the colum in the Targets table
+                    # However, this refers to kernel_version.version, not to kernel_version as a whole
+                    target_info.kernel_version.version,
+                    target_info.kernel_version.release,
+                    target_info.kernel_version.sha1,
+                    target_info.kernel_config,
+                    target_info.sched_features))
+            # ResourceGetters table (one time)
+            for resource_getter in run_output.run_config.resource_getters:
+                self.current_resource_getter_uuid = uuid.uuid4()
+                self.cursor.execute(
+                    self.sql_command['create_resource_getter'],
+                    (
+                        self.current_resource_getter_uuid,
+                        self.run_uuid,
+                        resource_getter))
+                self.insert_parameters(
+                    'resource_getter',
+                    run_output.run_config.resource_getters[resource_getter],
+                    owner_id=self.current_resource_getter_uuid,
+                    job_specific=False)
+            self.first_job_run = False
+
+        # See wa/framework/output.py
+        # Job events
```

Comment:
Each of these headings should be a separate method instead.

Reply:
Done